### PR TITLE
Supported ReduceSum op

### PR DESCRIPTION
**Pull request by l-bat**
relates: #14312
merge with extra: opencv/opencv_extra#787

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under OpenCV (BSD) License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [x] There is reference to original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

<cut/>

```
force_builders=Custom,Custom Win,Custom Mac
build_image:Custom=ubuntu-openvino-2020.4.0:16.04
build_image:Custom Win=openvino-2020.4.0
build_image:Custom Mac=openvino-2020.3.0

test_modules:Custom=dnn,python2,python3,java
test_modules:Custom Win=dnn,python2,python3,java
test_modules:Custom Mac=dnn,python2,python3,java

buildworker:Custom=linux-1
# disabled due high memory usage: test_opencl:Custom=ON
test_opencl:Custom=OFF
test_bigdata:Custom=1
test_filter:Custom=*
```

### CODE REVIEW

File: modules/dnn/src/tensorflow/tf_importer.cpp [hunk 1]
```diff
@@ -2092,72 +2092,105 @@ void TFImporter::populateNet(Net dstNet)
...
-                    std::string squeezeName = name + "/squeeze";
-                    CV_Assert(layer_id.find(squeezeName) == layer_id.end());
-                    squeezeLp.set("axis", 0);
-                    squeezeLp.set("end_axis", 1);
-                    int squeezeId = dstNet.addLayer(squeezeName, "Flatten", squeezeLp);
-                    layer_id[squeezeName] = squeezeId;
-                    connect(layer_id, dstNet, Pin(layerShapeName), squeezeId, 0);
-                    layerShapeName = squeezeName;
-                }
+                    LayerParams flattenLp;
```
Comment (dkurt) [inline modules/dnn/src/tensorflow/tf_importer.cpp:2100]: Is there a way to keep origin alignment?

Comment (dkurt) [reply modules/dnn/src/tensorflow/tf_importer.cpp:2100]: Just to minimize the changes

Comment (l-bat) [reply modules/dnn/src/tensorflow/tf_importer.cpp]: Done

**Comment by dkurt** [comment]
please take a look at failed tests: https://pullrequest.opencv.org/buildbot/builders/precommit_custom_linux/builds/4234

**Pull request merged**